ufcg/src/main/java/br/com/askufcg/utils/JwtUtils.java

```diff
@@ -0,0 +1,59 @@
+package br.com.askufcg.utils;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Component
+public class JwtUtils {
+    @Value("${jwt.secret}")
+    private String secret;
+
+    private static final Date TWO_HOURS = new Date(System.currentTimeMillis() + 60 * 60 * 1000 * 2);
+
+    public String extractUsername(String token) {
+        return extractClaim(token, Claims::getSubject);
+    }
+
+    public Date extractExpiration(String token) {
+        return extractClaim(token, Claims::getExpiration);
+    }
+
+    public <T> T extractClaim(String token, Function<Claims, T> claimsResolver) {
+        final Claims claims = extractAllClaims(token);
+        return claimsResolver.apply(claims);
+    }
+
+    private Claims extractAllClaims(String token) {
+        return Jwts.parser().setSigningKey(secret).parseClaimsJws(token).getBody();
+    }
+
+    private Boolean isTokenExpired(String token) {
+        return extractExpiration(token).before(new Date());
+    }
+
+    public String generateToken(String username) {
+        Map<String, Object> claims = new HashMap<>();
+        return createToken(claims, username);
+    }
+
+    private String createToken(Map<String, Object> claims, String subject) {
+
+        return Jwts.builder().setClaims(claims).setSubject(subject).setIssuedAt(new Date(System.currentTimeMillis()))
```

Comment:
Entendi que ele cria o token, mas não entendi como isso é feito. O que é claims e porque sempre um novo hashmap?

Reply:
Claims constitute the payload part of a JSON web token and represent a set of information exchanged between two parties

Reply:
https://learn.akamai.com/en-us/webhelp/api-gateway/api-gateway-user-guide/GUID-C495E5E1-B70E-4D30-BB8C-3707BAAAB778.html